inception/inception-log/src/main/java/de/tudarmstadt/ukp/inception/log/config/EventLoggingPropertiesImpl.java

```diff
@@ -55,14 +54,34 @@ public void setEnabled(boolean aEnabled)
     }
 
     @Override
-    public Set<String> getExcludeEvents()
+    public Set<String> getIncludePatterns()
+    {
+        return includePatterns;
+    }
+
+    @Override
+    public void setIncludePatterns(Set<String> includePatterns)
+    {
+        this.includePatterns = includePatterns;
+    }
+
+    @Override
+    public Set<String> getExcludePatterns()
+    {
+        return excludePatterns;
+    }
+
+    @Override
+    public void setExcludePatterns(Set<String> excludePatterns)
     {
-        return excludeEvents;
+        this.excludePatterns = excludePatterns;
     }
 
     @Override
-    public void setExcludeEvents(Set<String> aExcludeEvents)
+    public boolean shouldLogEvent(String eventName)
     {
-        excludeEvents = aExcludeEvents;
+        return includePatterns.stream().anyMatch(pattern -> Pattern.matches(pattern, eventName))
```

Comment:
This code compiles the patterns and then matches them against the names every time an event is processed. This is quite slow. Assuming the set of names of events is finite, I would suggest to set up a (static) cache where once you discover if a given event name should be logged or not, you enter that event name with a true/false value. Thus, you only need to check the patterns when an event name is not yet in the cache (`O(n)`). If it is in the cache, then the lookup is `O(1)`.